applications/istio/bookinfo/README.md

```diff
@@ -0,0 +1,130 @@
+# Istio Bookinfo demo in OAM way
+This application demonstrates how OAM bootstraps [Istio Bookinfo application](https://istio.io/latest/docs/examples/bookinfo/) with workload `Gateway`, `VirtualService`,
+`DestinationRule` deployed by OAM.
+
+## Prerequisites
+- Helm v3
+
+- OAM Kubernetes Runtime
+
+```
+kubectl create namespace crossplane-system
+helm repo add crossplane-alpha https://charts.crossplane.io/alpha
+helm install addon-oam-kubernetes-local --namespace crossplane-system crossplane-alpha/oam-core-resources
+```
+
+- OAM workloads/traits definitions
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/crossplane/crossplane/release-0.12/docs/snippets/run/definitions.yaml
+workloaddefinition.core.oam.dev/containerizedworkloads.core.oam.dev created
+traitdefinition.core.oam.dev/manualscalertraits.core.oam.dev created
+workloaddefinition.core.oam.dev/postgresqlinstancerequirements.database.example.org created
+
+```
+
+(TODO)  ** The root cause why this depends on installation of Crossplane has to be investigated.**
+```
+$ kubectl apply -f https://raw.githubusercontent.com/crossplane/crossplane/release-0.12/docs/snippets/run/definitions.yaml
+
+unable to recognize "https://raw.githubusercontent.com/crossplane/crossplane/release-0.12/docs/snippets/run/definitions.yaml": no matches for kind "WorkloadDefinition" in version "core.oam.dev/v1alpha2"
+unable to recognize "https://raw.githubusercontent.com/crossplane/crossplane/release-0.12/docs/snippets/run/definitions.yaml": no matches for kind "TraitDefinition" in version "core.oam.dev/v1alpha2"
+unable to recognize "https://raw.githubusercontent.com/crossplane/crossplane/release-0.12/docs/snippets/run/definitions.yaml": no matches for kind "WorkloadDefinition" in version "core.oam.dev/v1alpha2"
+```
+
+- Istio 1.6.3
+
+Pleaser refer to [Download and Install Istio](https://istio.io/latest/docs/setup/getting-started/).
+
+## Deploy the application
+
+```
+$ kubectl apply -f bookinfo.yaml
+```
+(TODO) Not sure whether `Service` and `Deployment` from [bookinfo.yaml](./bookinfo.yaml) should be converted to
```

Comment:
I would suggest to use the components. And please remove the bookinfo.yaml.

Reply:
Is this way, the appconfig file will become so long. Still hesitating.

Reply:
Service is trait and Deployment is workload, let's keep everything straightforward (i.e. use OAM as k8s app definition only) for the first step and think about abstractions (`ContainerizedWorkload` etc) later. For abstractions, it should be more than a simple ContainerizedWorkload - new Trait and Scope abstractions are needed. 

@zzxwill In the context above, you don't even need to install `addon-oam-kubernetes-local`. But you need to define/generate definition objects for your workload and trait.